lib/Agrammon/DataSource/JSON.pm6

```diff
@@ -0,0 +1,54 @@
+use v6;
+use JSON::Fast;
+
+use Agrammon::Inputs;
+use Agrammon::DataSource::Util;
+
+class Agrammon::DataSource::JSON {
+    method load($simulation-name, $dataset-id, $json-data) {
+        my %input-data = from-json $json-data;
+        my $inputs = Agrammon::Inputs.new(:$simulation-name, :$dataset-id);
+        for %input-data.kv -> $full-tax, $module-data {
+            if $full-tax.index('[') -> $sub-start {
+                my $tax = $full-tax.substr(0, $sub-start);
+                my $sub-end = $full-tax.index(']');
+                my $sub-tax = $full-tax.substr($sub-end + 1);
+                $sub-tax = $sub-tax ?? $sub-tax.substr(2) !! '';
+                for $module-data.kv -> $instance, $instance-inputs {
+                    for $instance-inputs.kv -> $var, $value {
+                        $inputs.add-multi-input(
+                                $tax, $instance, $sub-tax,
+                                $var, maybe-numify($value)
```

Comment:
Do we need to `maybe-numify` in the JSON? Since it has a concept of strings vs. numbers, perhaps we just expect that numeric values are property provided that way?

Reply:
Removed.